src/tools/auth0/handlers/themes.ts

```diff
@@ -0,0 +1,656 @@
+import { cloneDeep } from 'lodash';
+import { Assets } from '../../../types';
+import log from '../../../logger';
+import DefaultHandler from './default';
+
+export default class ThemesHandler extends DefaultHandler {
+  existing: ThemeRequest[] | null;
+
+  constructor(options: DefaultHandler) {
+    super({
+      ...options,
+      type: 'themes',
+    });
+  }
+
+  async getType(): Promise<ThemeRequest[] | null> {
+    if (!this.existing) {
+      this.existing = await this.getThemes();
+    }
+
+    return this.existing;
+  }
+
+  async getThemes(): Promise<ThemeRequest[] | null> {
+    let theme;
+
+    try {
+      theme = await this.client.branding.getDefaultTheme();
+    } catch (err) {
+      // Errors other than 404 (theme doesn't exist) or 400 (no-code not enabled) shouldn't be expected
+      if (err.statusCode !== 404 && err.statusCode !== 400) {
+        throw err;
+      }
+    }
+
+    if (theme) {
+      delete theme.themeId;
+      return [theme];
+    }
+
+    return null;
+  }
+
+  async processChanges(assets: Assets): Promise<void> {
+    const { themes } = assets;
+
+    if (!themes || themes.length === 0) {
+      return;
+    }
+
+    if (themes.length > 1) {
+      log.warn('Only one theme is supported per tenant');
```

Comment:
Couldn't add a test for this case because we aren't using `proxyquire` or a similar dependency replacement library